File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> temp;
        int l=0, f=0;
        int maxLen=0;
        for(int r=0; r<s.size(); r++){
            temp[s[r]]++;
            f = max(f, temp[s[r]]);
            while((r-l+1)-f>k){
                temp[s[l]]--;
                l++;
            }
            maxLen = max(maxLen, r-l+1);
        }
        return maxLen;
    }
};

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-1.py
class Solution:
    def shipWithinDays(self, weights: List[int], days: int) -> int:
        l=max(weights)
        r = sum(weights)
        res = r
        while l<=r:
            m = l + ((r-l)//2)
            count = 0
            temp = 0
            for weight in weights:
                temp += weight
                if temp > m:
                    count+=1
                    temp = weight
            while temp>m:
                count+=1
                temp = temp-m
            count+=1
            print(count)
            if count > days:
                l = m+1
            else:
                r = m-1
                res = m
                
        return res